### PR TITLE
feat(snes): general-purpose DMA with cycle accounting (#2745)

### PR DESCRIPTION
**Pull request by rmstdope**
## Summary
- add `src/snes/bus/dma.rs` with a dedicated `DmaController` and channel register model (`$43x0-$43xB`)
- implement synchronous MDMAEN (`$420B`) execution in `SnesSystemBus` for channels 0..7 in priority order
- implement transfer modes 0..4 plus canonical alias mapping for 5/6/7 -> 1/2/3
- implement DMA A-bus address stepping (increment/decrement/fixed), 16-bit wrap semantics, and `DAS=$0000 => 65536 bytes`
- enforce DMA-specific A-bus MMIO exclusion/open-bus behavior for `$2100-$21FF`, `$4000-$41FF`, `$4200-$421F`, `$4300-$437F`
- wire deterministic cycle accounting: `8 cycles/byte + 8/channel + 16 global`
- add/extend SNES bus tests for synchronous DMA behavior, multi-channel ordering, mode alias behavior, MMIO exclusion, count-zero behavior, and register-block non-aliasing
- update `architecture.md` SNES section with DMA module/status details

## Testing
- `cargo fmt --all`
- `cargo clippy --all-targets --all-features -- -D warnings`
- `./scripts/test-dir.sh src/snes --skip-integration`
- `cargo test --no-default-features --lib`
- `wasm-pack test --headless --chrome --no-default-features --features wasm`
- `source .venv/bin/activate && python -m unittest discover -s scripts -t scripts -p "test_*.py" && python -m unittest discover -s scripts/metadata-scraper -t scripts/metadata-scraper -p "test_*.py" && python -m unittest discover -s scripts/nes-rom-db-scraper -t scripts/nes-rom-db-scraper -p "test_*.py"`
- `npm test`

Closes #2745

### CODE REVIEW

File: src/snes/bus/system_bus.rs [hunk 1]
```diff
@@ -154,6 +155,62 @@ impl SnesSystemBus {
...
+    }
+
+    fn start_dma_transfer(&mut self, mdmaen: u8) {
+        let mut dma = std::mem::take(&mut self.dma);
+        let consumed_ticks = dma.start_dma(mdmaen, self, self.mdr.get());
+
+        self.ticks
+            .set(self.ticks.get().wrapping_add(consumed_ticks));
+        self.dma = dma;
+    }
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/snes/bus/system_bus.rs:213]: Fixed in 19c787ad. DMA now returns the final open-bus byte, and `SnesSystemBus` writes it back into `mdr` after transfer completion. Added regression test `dma_updates_mdr_with_last_transferred_byte` to lock behavior.

File: src/snes/bus/dma.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            if (mdmaen & (1 << channel)) == 0 {
+                continue;
+            }
+
+            ticks += 8;
+            ticks += self.run_channel(channel, abus, &mut open_bus);
+        }
+
+        ticks
+    }
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/snes/bus/dma.rs]: Fixed in 19c787ad. DMA now returns the final open-bus byte, and `SnesSystemBus` writes it back into `mdr` after transfer completion. Added regression test `dma_updates_mdr_with_last_transferred_byte` to lock behavior.
